### PR TITLE
[Chi 383] redesign alert messages

### PR DESCRIPTION
**Pull request by janibolkvadze**
@jllr @mattnickles

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/319/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/320/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/321/ for more details.

**Comment by janibolkvadze** [comment]
Thanks for suggestions!
Almost all done.

- [x] change .m-alert font-size to base body font-size: $text (14px - 0.875rem)

- [x] remove size classes from all icons in the alert examples except -lg so they default to 16px

- [x] sm alert height should be 48px (to achieve use padding: 0.813rem 1rem;)

- [x] md alert height should be 56px (to achieve use padding: 1.063rem 1rem;)

- [x] lg alert height should be 80px with title/heading and alert 

- [x] the title/heading in -lg alerts is missing a 4px (0.25rem) margin-bottom

- [x] the top value on icons within -lg alerts doesn't behave well when a title/heading isn't included
Screenshot 2019-10-16 at 15 42 23
This is the current approach how .a-icon and title/heading are aligned in -lg alerts. But If title/heading is not provided the property top: 0.1875rem breaks the positioning of the icon. I suggest to introduce new class like "-titled" for the -lg alerts with title and apply this property (top: 0.1875rem;) to the icon only if this class is present. I implemented this approach.
Not sure about the class name. What do you think about it?

- [] the alert text in -lg alerts is not respecting the parent 20px (1.25rem) line-height
Alert text has line-height: 20px, title/heading line-height is 24px. Should I make title line-height 20px?

I also discovered an issue with Toast alerts:
In tests/components/alert-toast.html you can see that there are some cases when we use toast titles of different sizes (h1-h6). The problem is that in this case, we can not align icon using 'top'  property. And we also need to think about icon sizing, because besides H1, H2, H3 it looks tiny
Screenshot 2019-10-16 at 19 42 40

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/323/ for more details.

File: src/chi/components/alert/_alert-common.scss [hunk 1]
```diff
@@ -64,8 +74,28 @@ $states: (
 
   @each $state in map-keys($states) {
     &.-#{$state} {
-      background-color: map-get(map-get($states, $state), background-color);
-      border-color: map-get(map-get($states, $state), border-color);
+      border-left: none;
+      color: set-color(grey, 100);
+      &.-#{$state}.-bubble::before, &.-#{$state}.-toast::before {
```
Comment (jllr) [inline src/chi/components/alert/_alert-common.scss]: &.-#{$state}.-toast::before should be in another line for better code readability

File: src/chi/components/alert/_alert-common.scss [hunk 2]
```diff
@@ -64,8 +74,28 @@ $states: (
...
+        background-color: map-get(map-get($states, $state), border-left-color);
+      }
+      &.-#{$state}.-banner {
+        background-color: map-get(map-get($states, $state), banner-color);
+        border: none;
+        border-top: 1px solid transparent;
+        border-bottom: 1px solid map-get(map-get($states, $state), banner-border-color);
+        box-shadow: none;
+      }
+      &.-#{$state} .a-h5 {
```
Comment (jllr) [inline src/chi/components/alert/_alert-common.scss]: &.-#{$state} is not necessary as it is in line 76

File: src/chi/components/alert/_alert-common.scss [hunk 3]
```diff
@@ -32,14 +42,14 @@ $states: (
...
   border-width: 0.0625rem;
-  box-sizing: border-box;
   color: $text-color;
   display: flex;
   font-family: $font-family;
-  font-size: $text-small;
+  font-size: $text;
   line-height: $line-height-small;
-  padding: 0.5625rem 1rem;
+  padding: 0.8133rem 1rem;
```
Comment (jllr) [inline src/chi/components/alert/_alert-common.scss]: padding should be the one from -md as it is the base alert

File: src/website/views/components/alert.pug [hunk 4]
```diff
@@ -13,113 +13,219 @@ h4 Base
 .example.-mb--3
   .-p--1
     each type in ['success', 'danger', 'warning', 'info', 'muted']
-      .m-alert.-bubble.-m--2(class=`-${type}`)
+      .m-alert.-bubble.-m--2.-md(class=`-${type}`)
```
Comment (jllr) [inline src/website/views/components/alert.pug]: Base alert is -md, it should work without the class -md. When we remove this class the alerts get the -sm size which is incorrect. Please change it in all the examples.

File: src/website/views/components/alert.pug [hunk 5]
```diff
@@ -13,113 +13,219 @@ h4 Base
 .example.-mb--3
   .-p--1
     each type in ['success', 'danger', 'warning', 'info', 'muted']
```
Comment (jllr) [inline src/website/views/components/alert.pug]: Examples in Base are going to be now the Alerts with icons, please change and remove the Icons section

File: src/website/views/components/alert.pug (hunk 4 above)
Comment (mattnickles) [reply src/website/views/components/alert.pug]: @janibolkvadze In addition to this, the base alert type is -bubble. So if the bubble class is removed, it should still render as a bubble alert. We have a lot of alerts in production today without the -bubble class that will break with this PR. Here's an example of one on the grid page. Notice the border isn't rendering and the size doesn't match md
image

**Comment by mattnickles** [comment]
@janibolkvadze @jllr Let's hold off on making any changes to this. Working on an updated spec that I will have ready tomorrow.

**Comment by mattnickles** [comment]
Closing and opening new PR with changes

**Pull request closed without merging**